ui/app/components/totp/key-create-toggle-groups.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class KeyCreateToggleGroupsComponent extends Component {
+  @tracked showGroup = null;
+
+  get groups() {
+    const { generate } = this.args.model;
+
+    const groups = {
+      'TOTP Code Options': ['algorithm', 'digits', 'period'],
+    };
+
+    if (generate) {
+      groups['Provider Options'] = ['key_size', 'skew', 'exported', 'qr_size'];
+    }
+
+    return groups;
+  }
+
+  @action
+  toggleGroup(group, isOpen) {
+    this.showGroup = isOpen ? group : null;
+  }
```

Comment:
This is great! Much easier to read instead of being in the model. Since this logic is fairly minimal (and the template .hbs file isn't huge), I think this could still live directly in `key-create` but we can keep it separate if you like having `KeyCreateToggleGroups` as a separate component!